leetcodePython/SlidingWindow/slidingwindow_862.py

```diff
@@ -0,0 +1,25 @@
+from typing import List
+from collections import deque
+
+
+class Solution:
+    def shortestSubarray(self, nums: List[int], k: int) -> int:
+        sums = [0]
+        for x in nums:
+            sums.append(sums[-1] + x)
+        monosum_index_q = deque()
+        ans = len(nums) + 1
+        for right, sum in enumerate(sums):
+            while monosum_index_q and sums[monosum_index_q[-1]] >= sum:
+                monosum_index_q.pop()
+            while monosum_index_q:
+                left = monosum_index_q[0]
+                if sum - sums[left] < k:
+                    break
```

Comment:
Sum is increasing. So we can shrink the left window to make the window smaller. Thus getting the smallest sub array

Reply:
monotonic increase of `sum[left]` leads to monotonic decrease of remaining. And that ensures our `sum - sums[left] < k` is a valid check (since it decrease)